src/main/java/io/skai/okta/internshipstreamingserviceparserxml/services/converter/EpisodeConverter.java

```diff
@@ -4,16 +4,25 @@
 import io.skai.okta.internshipstreamingserviceparserxml.dto.LostFilmRssItem;
 import org.springframework.stereotype.Component;
 
+import java.security.InvalidParameterException;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+
 @Component
 public class EpisodeConverter {
+    private final static Pattern TVS_TITLE_PATTERN = Pattern.compile("(?<=\\().+?(?=\\))");
+    private final static Pattern SEASON_NUMBER_PATTERN = Pattern.compile("(?<=\\(S)\\d{2}(?=E\\d{2})");
+    private final static Pattern EPISODE_NUMBER_PATTERN = Pattern.compile("(?<=\\(S\\d{2}E)\\d{2}(?=\\))");
+
     public Episode map(LostFilmRssItem lostFilmRssItem) {
         String title = lostFilmRssItem.getTitle();
-        int season = Integer.parseInt(extract(title, "season"));
-        int episodeNumber = Integer.parseInt(extract(title, "episode"));
-        String tvSeriesTitle = extract(title, "tvsTitle");
+
+        String tvSeriesTitle = extract(title, EpisodeField.TV_SERIES_TITLE);
```

Comment:
you can call here extractField method directly and remove completely extract method with ugly switch case (hate them)

Reply:
ok. but it still would be situations, when we need use switch-case?

Reply:
better to avoid it, there are plenty more nicer ways/patters how to do it